main.go

```diff
@@ -0,0 +1,272 @@
+	"github.com/urfave/cli/v2"
+)
+
+// Define flag names as constants
+const (
+	FlagEnv                       = "env"
+	FlagBidderAddress             = "bidder-address"
+	FlagUsePayload                = "use-payload"
+	FlagRpcEndpoint               = "rpc-endpoint"
+	FlagWsEndpoint                = "ws-endpoint"
+	FlagPrivateKey                = "private-key"
+	FlagOffset                    = "offset"
+	FlagBidAmount                 = "bid-amount"
+	FlagBidAmountStdDevPercentage = "bid-amount-std-dev-percentage"
+	FlagNumBlob                   = "num-blob"
+	FlagDefaultTimeout            = "default-timeout"
+)
+
+func main() {
+	// Initialize the slog logger with JSON handler and set log level to Info
+	logger := slog.New(slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
+		Level:      slog.LevelInfo,
+		AddSource:  true,
+	}))
+	slog.SetDefault(logger)
+
+	app := &cli.App{
+		Name:  "Preconf Bidder",
+		Usage: "A tool for bidding in mev-commit preconfirmation auctions for blobs and transactions",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    FlagEnv,
+				Usage:   "Path to .env file",
+				EnvVars: []string{"ENV_FILE"},
+			},
+			&cli.StringFlag{
+				Name:    FlagBidderAddress,
+				Usage:   "Address of the bidder",
+				EnvVars: []string{"BIDDER_ADDRESS"},
+				Value:   "mev-commit-bidder:13524",
+			},
+			&cli.BoolFlag{
+				Name:    FlagUsePayload,
+				Usage:   "Use payload for transactions",
+				EnvVars: []string{"USE_PAYLOAD"},
+				Value:   true,
+			},
+			&cli.StringFlag{
+				Name:     FlagRpcEndpoint,
+				Usage:    "RPC endpoint when use-payload is false",
+				EnvVars:  []string{"RPC_ENDPOINT"},
+				Required: false,
+			},
+			&cli.StringFlag{
+				Name:     FlagWsEndpoint,
+				Usage:    "WebSocket endpoint for transactions",
+				EnvVars:  []string{"WS_ENDPOINT"},
+				Required: true,
+			},
+			&cli.StringFlag{
+				Name:      FlagPrivateKey,
+				Usage:     "Private key for signing transactions",
+				EnvVars:   []string{"PRIVATE_KEY"},
+				Required:  true,
+				Hidden:    true,
+				TakesFile: false,
+			},
+			&cli.Uint64Flag{
+				Name:    FlagOffset,
+				Usage:   "Offset value for transactions",
+				EnvVars: []string{"OFFSET"},
+				Value:   1,
+			},
+			&cli.Float64Flag{
+				Name:    FlagBidAmount,
+				Usage:   "Amount to bid",
+				EnvVars: []string{"BID_AMOUNT"},
+				Value:   0.001,
+			},
+			&cli.Float64Flag{
+				Name:    FlagBidAmountStdDevPercentage,
+				Usage:   "Standard deviation percentage for bid amount",
+				EnvVars: []string{"BID_AMOUNT_STD_DEV_PERCENTAGE"},
+				Value:   100.0,
+			},
+			&cli.UintFlag{
+				Name:    FlagNumBlob,
+				Usage:   "Number of blobs to send (0 for ETH transfer)",
+				EnvVars: []string{"NUM_BLOB"},
+				Value:   0,
+			},
+			&cli.UintFlag{
+				Name:    FlagDefaultTimeout,
+				Usage:   "Default timeout in seconds",
+				EnvVars: []string{"DEFAULT_TIMEOUT"},
+				Value:   15, // Default to 15 seconds
+			},
+		},
+		Action: func(c *cli.Context) error {
```

Comment:
The `c.Context` should be used as parent context in all places where you create a new context. This will help you shutdown the app gracefully.